js/plugin/connection.ts

```diff
@@ -6,6 +6,7 @@ export const ConnectionMixin = (SuperClass) => {
     public connection;
     private _data;
     public connected = false;
+    public ready = false;
```

Comment:
Does `ready` need to be public? Can it be private? Also, without knowing the WS API, a future maintainer may not understand what `ready` means without further code inspection. I suggest renaming to `private _wsConnected` or something similar. I will be making similar suggestions so the code reads well to what is to do with wsConnection and what is to do with component connection.

Reply:
As per later comments, check on ready vs connected as to what sits best for future maintainers to easily debug. It may be that even `connected` gets renamed to `wsConnected`.